packages/datadog-plugin-kafkajs/test/index.spec.js

```diff
@@ -184,6 +184,29 @@ describe('Plugin', () => {
 
             return expectedSpanPromise
           })
+
+          it('should run constructor even if no eachMessage supplied', (done) => {
+            let eachBatch = async ({ batch }) => {
+              try {
+                expect(batch.isEmpty()).to.be.false
+                done()
+              } catch (e) {
+                done(e)
+              } finally {
+                eachBatch = () => {} // avoid being called for each message
+              }
+            }
+
+            const runResult = consumer.run({ eachBatch: (...args) => eachBatch(...args) })
+
+            if (!runResult || !runResult.then) {
```

Comment:
The old version actually trips here already, because the patched `run` method returns a plain object under these circumstances instead of the expected promise.